addons/survey/controllers/main.py

```diff
@@ -311,10 +311,29 @@ def _prepare_survey_data(self, survey_sudo, answer_sudo, **post):
     def _prepare_question_html(self, survey_sudo, answer_sudo, **post):
         """ Survey page navigation is done in AJAX. This function prepare the 'next page' to display in html
         and send back this html to the survey_form widget that will inject it into the page."""
-        data = self._prepare_survey_data(survey_sudo, answer_sudo, **post)
+        survey_data = self._prepare_survey_data(survey_sudo, answer_sudo, **post)
         if answer_sudo.state == 'done':
-            return request.env.ref('survey.survey_fill_form_done').render(data)
-        return request.env.ref('survey.survey_fill_form_in_progress').render(data)
+            result = {'survey_content': request.env.ref('survey.survey_fill_form_done').render(survey_data)}
+        else:
+            result = {'survey_content': request.env.ref('survey.survey_fill_form_in_progress').render(survey_data)}
+
+        if answer_sudo.state == 'in_progress' and not survey_data.get('question', request.env['survey.question']).is_page:
+            if survey_sudo.questions_layout == 'page_per_section':
+                page_ids = survey_sudo.page_ids.ids
+                result['survey_progress'] = request.env.ref('survey.survey_progression').render({
+                    'survey': survey_sudo,
+                    'page_ids': page_ids,
```

Comment:
[C5] We could directly prepare the dict, and include the condition like
```
'page_ids': survey_sudo.page_ids.ids if survey_sudo.questions_layout == 'page_per_section' else survey_sudo.question_ids.ids
'page_number': blah
```

That way we have less embedded if / else .

Reply:
I like the idea, but I tried and it's actually a lot less readable.
Mainly because of the fact that we re-use page_ids for the page_number.